recipes/sipbuildtool/conanfile.py

```diff
@@ -0,0 +1,140 @@
+import os
+import sys
+
+from pathlib import Path
+from io import StringIO
+
+from jinja2 import Template
+
+from conan import ConanFile
+from conan.tools.env import VirtualRunEnv
+from conan.tools.files import files
+from conans.errors import ConanException
+
+
+class SipBuildTool(object):
+    """
+    A generator that creates a CMake script to build python bindings for
+    projects using Sip.
+
+    This generator expects the project to have a pyproject.toml.jinja file that
+    determines which Sip files to compile. It configures that file to a TOML
+    file that Sip can use.
+
+    Usage in a conanfile for a Sip project:
+    sip = self.python_requires["sipbuildtool"].module.SipBuildTool(self)
+    sip.configure()
+    sip.generate("projectName")
+    """
+    def __init__(self, conanfile: ConanFile):
+        self.conanfile = conanfile
+        env = VirtualRunEnv(self.conanfile)
+        self.run_env = env.environment()
+        self.envvars = None
+        self.python_venv_interpreter = None
+        self.sip_build_executable = None
```

Comment:
For these python properties I like for them to be typed as it makes it easier to read/maintain

```suggestion
        self.conanfile: ConanFile = conanfile
        env = VirtualRunEnv(self.conanfile)
        self.run_env: Environment = env.environment()
        self.envvars: Optional[EnvVars] = None
        self.python_venv_interpreter: Optional[Path] = None
        self.sip_build_executable: Optional[Path] = None
```

Reply:
Yes agree, for developers that wrote a dependency manager for a strong type language C++, they sure don't use `typing` something I followed along with (mostly, but not consistently) since I wanted to mimic their style as much as possible to not introduce unknown behavior.
Which is a real possibility since they do some wonkey-things with inheritance with Conan python_requires.

But I'm confident enough to allow for these typings, provided that the imports at the top of this files are also there